### DISCUSSION TITLE
Turn inverter on and off?

### DISCUSSION THREAD

**Discussion by rpdrewes** [jblance/mpp-solar#374] [Q&A]
Hello all,

First, thank you to developers for this work. I took a while to find something that would allow me to get important information from my EG4 6500EXs without dealing with closed source nonsense, and this has worked very well!

Question: is there any facility to turn inverters on and off (turn loads on and off) from any of these protocols supported by this code? I looked at the command set for several protocols and I do not see anything obvious.

With my Outback Skybox I can turn load (inverter) on and off remotely via their OpticsRE interface but never did discover how to do it through the SunSpec python interface I hacked up.  I'm wondering if there is some proper way to do it via this code talking to (for example) EG4 6500EXs.

Thanks!

**Comment by jblance** [comment]
If the protocol supports on/off and the protocol is implemented for
mppsolar then yes.

What protocol is used?
Is the inverter currently supported with this code?

**Reply by rpdrewes** [reply]
I get the device ID:

> # mpp-solar -p /dev/ttyUSB0 --getDeviceId
> 
> Command: Get Device ID - Generate a device id
> -------------------------------------------------------------------------------- 
> Parameter Value             Unit
> deviceid  PI30:045:MKS2-6500   

From the wiki, the protocol for this device should be PI30MAX.

I can get the list of commands for that protocol with -c but none of the commands looks like it is the right one to turn the inverter on and off (set operating mode to Standby or Online). Maybe POPLG?

If someone can suggest a command that works on some other inverter I can maybe get an idea of what might work for this one. Looking at the EG4 6500EX modbus protocol document here https://eg4electronics.com/wp-content/uploads/2022/09/Modbus-protocol-EG4-6500-EX.docx.pdf I am starting to think that there is no way to turn the inverter off remotely ...

**Comment by jblance** [comment]
Oh, yeah i have the same protocol and problem
There is prolly some undocumented command
or using a comnbination of `PBATCD<abc>` / `PBATMAXDISC<nnn>`
